src/m1_facilitator/models/Message.ts

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+import BigNumber from 'bignumber.js';
+
+import Comparable from '../../m0_facilitator/observer/Comparable';
+
+/**
+ * Represents Message model object.
+ */
+export default class Message extends Comparable<Message> {
+  public messageHash: string;
+
+  public type: string;
+
+  public intentHash: string;
```

Comment:
`intentHash` is optional
```suggestion
  public intentHash?: string;
```

Reply:
Done.